### PR TITLE
add onkrakenmount property; removed node v4, v6 and added node 10 in travis CI config

### PR DESCRIPTION
**Pull request by kumarrishav**
moving the implementation details from #513 

This will let have modules the app instance once Kraken (express sub-app) mounts.

### CODE REVIEW

**Comment by kumarrishav** [comment]
@tlivings @aheckmann @grawk @shaunwarman review?

File: lib/bootstrap.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ var debug = require('debuglog')('kraken/bootstrap');
 
 module.exports = function (app, options) {
 
+    // on bootstrapinit
+    const onbootstrapinit = options.onbootstrapinit;
```
Comment (sumeetkakkar) [inline lib/bootstrap.js]: My understanding is that this is invoked when kraken is mounted in an express app and express app instance is available to pass down the callback function. Thoughts on calling this as `onkrakenmount` or `onmount`?

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 language: node_js
 node_js:
-  - "4"
   - "6"
   - "8"
+  - "10"
```
Comment (grawk) [inline .travis.yml:4]: since you're cleaning up, is there a reason not to remove "6"?

**Review by grawk — CHANGES_REQUESTED** [review]
No issues with the code change. Would just recommend to remove node 6 from the travis config.

File: test/kraken.js [hunk 3]
```diff
@@ -178,6 +178,59 @@ test('kraken', function (t) {
...
+            t.pass('server started');
+        }
+
+        function error(err) {
+            t.error(err, 'server startup failed');
+        }
+
+        options = {
+            onkrakenmount: function (app) {
+                const serverListen = app.listen;
```
Comment (sumeetkakkar) [inline test/kraken.js]: I think we should just assert that the app is an express instance.

File: test/kraken.js [hunk 4]
```diff
@@ -178,6 +178,31 @@ test('kraken', function (t) {
...
+            t.pass('bootstrap successful');
+        }
+
+        function error(err) {
+            t.error(err, 'bootstrap failed');
+        }
+
+        options = {
+            onkrakenmount: function (app) {
+                t.ok(app.mountpath, 'app instance exist');
```
Comment (sumeetkakkar) [inline test/kraken.js]: I think app would be a `function`. We can assert that as well.

**Review by grawk — APPROVED** [review]

**Review by sumeetkakkar — APPROVED** [review]

File: index.js [hunk 5]
```diff
@@ -57,6 +57,10 @@ module.exports = function (options) {
         // moved to `options` for use later.
         options.mountpath = app.mountpath;
 
+        // on krakenmount
+        const onkrakenmount = options.onkrakenmount;
```
Comment (Greg-Wilburn) [inline index.js]: Please use camel case for options and variables. This should be `onKrakenMount`

**Review by Greg-Wilburn — APPROVED** [review]

**Pull request merged**